app/auth/views/briefs.py

```diff
@@ -45,45 +107,39 @@ def get_brief_response(brief_response_id):
     return jsonify(briefResponses=brief_response.serialize())
 
 
-@auth.route('/brief/<int:brief_id>/respond', methods=["POST"])
-def post_brief_response(brief_id):
-    brief_response_json = get_json_from_request()
-
-    try:
-        brief = Brief.query.get(brief_id)
-    except DataError:
-        brief = None
-
-    if brief is None:
-        abort(400, "Invalid brief ID '{}'".format(brief_id))
+@auth.route('/brief/<int:brief_id>/respond/documents/<string:supplier_code>/<slug>', methods=['POST'])
+def upload_brief_response_file(brief_id, supplier_code, slug):
+    supplier, brief = _can_do_brief_response(brief_id)
+    return jsonify({"filename": s3_upload_file_from_request(request, slug,
+                                                            os.path.join(brief.framework.slug, 'documents',
+                                                                         'brief-' + str(brief_id),
+                                                                         'supplier-' + str(supplier.code)))
+                    })
 
-    if brief.status != 'live':
-        abort(400, "Brief must be live")
 
-    if brief.framework.status != 'live':
-        abort(400, "Brief framework must be live")
-
-    try:
-        supplier = Supplier.query.filter(
-            Supplier.code == current_user.supplier_code
-        ).first()
-    except DataError:
-        supplier = None
-
-    if not supplier:
-        abort(400, "Invalid supplier Code '{}'".format(current_user.supplier_code))
+@auth.route('/brief/<int:brief_id>/respond/documents/<int:supplier_code>/<slug>', methods=['GET'])
```

Comment:
will the buyer be able to retrieve these using existing "download responses" code?

Reply:
will need to modify supplier app in future to check both new and old locations